active/000-hive-lineage/000-hive-lineage.md

```diff
@@ -0,0 +1,460 @@
+
+4. **Troubleshooting:** By tracking data movement and transformations, Hive Lineage aids in troubleshooting data quality issues and anomalies. It provides insights into where and how data issues may have occurred, enabling quicker resolution and maintaining data integrity.
+
+5. **Data Quality Monitoring:** Hive Lineage helps identify data quality issues by providing visibility into data transformations and movements. This enables proactive monitoring and management of data quality, ensuring that data remains accurate and reliable.
+
+### Expected Outcome:
+
+By implementing Hive Lineage, we expect to achieve the following outcomes:
+
+- Improved data governance and control over data assets.
+- Enhanced compliance with regulatory requirements.
+- Better understanding of data flow and dependencies within the Hive ecosystem.
+- Effective impact analysis for schema, table, and transformation changes.
+- Quicker resolution of data quality issues and anomalies.
+- Overall improvement in data accuracy and reliability.
+
+## Detailed Design
+
+The design for Hive Lineage involves leveraging Hive's LineageLogger Hook to capture lineage information, processing this information, and sending it to a Kafka topic. A Kafka consumer then processes these messages and sends metadata change proposals (MCPs) to the GMS (Graph Metadata Service). This design ensures efficient tracking and documentation of data flow within the Hive ecosystem.
+
+### Implementation Details:
+
+1. **LineageLogger Hook:**
+    - We configure Hive to use the LineageLogger Hook by updating the Hive configuration to include this hook via client JAR. The JAR is registered in HiveServer to ensure it is invoked after query execution is finished and before the query result is published to the user. This ensures that the hook is invoked to capture lineage information after the query execution is completed ensuring the correct lineage is captured and doesn't get affected if the query execution fails.
+
+2. **Hive Lineage Hook (Thin Client):**
+    - **Extract Lineage:** The thin client, developed as a JAR, uses Hive hooks to capture lineage information from executed queries. Implement the thin client by extending the Hive hook interface. Capture events related to query execution, table creation, and data insertion.
+    - **JSON:** The captured lineage information is converted to JSON format for consistency and ease of processing.
+    - **Publish:** The JSON-formatted lineage information is sent to a predefined Kafka topic. This ensures that lineage data is reliably transmitted to downstream consumers.
+
+3. **Kafka Consumer:**
+    - The Kafka consumer listens to the predefined Kafka topic for incoming messages containing lineage information.
```

Comment:
Would the kafka topic with lineage information from Hive be managed by DataHub or would we assume that this topic exists independently? If DataHub is responsible, don't forget about having to support the schema in the internal schema registry.